options/create.go

```diff
@@ -137,6 +139,26 @@ func (opts *Create) Hash() hash.Hash {
 	return hash
 }
 
+func (opts *Create) resolveRemote(env []string) {
+	if opts.RemoteInfo == nil {
+		return
+	}
+
+	var remoteCmd string
+
+	if opts.WorkingDirectory != "" {
+		remoteCmd += fmt.Sprintf("cd '%s' && ", opts.WorkingDirectory)
+	}
+
+	if len(env) != 0 {
+		remoteCmd += strings.Join(env, " ") + " "
```

Comment:
Same question as for WorkingDirectory, but this time for Environment.

Reply:
I think it's fine...

Reply:
This will cause the ssh command to fail unless both the local filesystem and the remote one have this directory (e.g. if you want to use `/home/user` as the remote working directory, `/home/user` will have to exist on the local machine as well).

Also, if we're going to keep this overloaded `WorkingDirectory`, you have to modify `Create.Validate()` to not stat the WorkingDirectory in the remote case.